util/reggen/gen_dv.py

```diff
@@ -72,7 +105,44 @@ def gen_core_file(outdir: str,
         yaml.dump(core_data, core_file, encoding='utf-8')
 
 
-def gen_dv(block: IpBlock, dv_base_prefix: str, outdir: str) -> int:
+def get_dv_base_names_objects(dv_base_names: str) -> List[DvBaseNames]:
+    ''' Convert the dv_base_names from string to a list of DvBaseNames objects.'''
+    dv_base_names_list = []  # Output of this method
+    dv_base_names_list.append(DvBaseNames("default"))
+
+    if dv_base_names == "dv_base":
+        return dv_base_names_list
+
+    # Customerized dv_base_names comes with format:
+    # --dv-base-names "block-1:type:entity-name,block-2:type:entity-name"
+    for block_dv_base_names in dv_base_names.split(","):
+        items = block_dv_base_names.split(":")
+        block_name = items[0]
+        entity_type = items[1]
+        entity_name = items[2]
+
+        dv_base_names_list = [item for item in dv_base_names_list if item.ip == block_name]
+        if dv_base_names_list == []:
+            dv_base_names = DvBaseNames(block_name)
+            dv_base_names_list.append(dv_base_names)
+        else:
+            dv_base_names = dv_base_names_list[0]
+        dv_base_names.update_entity(entity_type, entity_name)
+    return dv_base_names_list
+
+
+def get_block_base_name(dv_base_names: str, block_name: str) -> DvBaseNames:
```

Comment:
I might misunderstand your previous comments regarding not storing the list of dv_base_names :) But with this method, we can directly call this function from `gen_dv` and `top_gen`, so I think it might be user to use. Please let me know if you think it is not so good :)

Reply:
See comment below: https://github.com/lowRISC/opentitan/pull/9007#discussion_r743097633
Its not clear to be why is this function called in `gen_dv`. It should be called directly in the template (as you are already doing). The issue is in the template, this function is called inside for loop that loops over each block. Which means it is constructing the dict of DvBaseNames internally again and again.